### DISCUSSION TITLE
ModuleNotFoundError: No module named 'torch'

### DISCUSSION THREAD

**Discussion by oksoil** [coolzhao/Geo-SAM#54] [General]
I am having trouble installing the plugin; any help here is appreciated: 

I am on  Apple M2 OSX Sequoia  15.1

This is the error I am getting: 

> Couldn't load plugin 'Geo SAM' 
> 
> ModuleNotFoundError: No module named 'torch' 
> Traceback (most recent call last):
>   File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 404, in loadPlugin
>     __import__(packageName)
>   File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 966, in _import
>     mod = _builtin_import(name, globals, locals, fromlist, level)
>   File "/Users/me/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/Geo SAM/__init__.py", line 7, in 
>     from .geo_sam_tool import Geo_SAM
>   File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 966, in _import
>     mod = _builtin_import(name, globals, locals, fromlist, level)
>   File "/Users/me/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/Geo SAM/geo_sam_tool.py", line 24, in 
>     from .tools.SAMTool import SAM_Model
>   File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 966, in _import
>     mod = _builtin_import(name, globals, locals, fromlist, level)
>   File "/Users/me/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/Geo SAM/tools/SAMTool.py", line 12, in 
>     from torch.utils.data import DataLoader
>   File "/Applications/QGIS.app/Contents/MacOS/../Resources/python/qgis/utils.py", line 966, in _import
>     mod = _builtin_import(name, globals, locals, fromlist, level)
> ModuleNotFoundError: No module named 'torch'
> 
> 
> Python version: 3.9.5 (default, Sep 10 2021, 16:18:19) [Clang 12.0.5 (clang-1205.0.22.11)] 
> QGIS version: 3.40.0-Bratislava Bratislava, 1d561ad8ad9 
> 
> Python Path:
> /Applications/QGIS.app/Contents/MacOS/../Resources/python
> /Users/me/Library/Application Support/QGIS/QGIS3/profiles/default/python
> /Users/me/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins
> /Applications/QGIS.app/Contents/MacOS/../Resources/python/plugins
> /Applications/QGIS.app/Contents/MacOS/lib/python39.zip
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/lib-dynload
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/pyproj-3.2.0-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/numpy-1.20.1-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/Rtree-0.9.7-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/cftime-1.2.1-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/GDAL-3.3.2-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/pandas-1.3.3-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/patsy-0.5.1-py3.9.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/Pillow-7.2.0-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/Fiona-1.8.13.post1-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/matplotlib-3.3.0-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/netCDF4-1.5.4-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/rasterio-1.1.5-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/scipy-1.5.1-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/numba-0.50.1-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/opencv_contrib_python-4.3.0.36-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/statsmodels-0.11.1-py3.9-macosx-10.13.0-x86_64.egg
> /Applications/QGIS.app/Contents/MacOS/lib/python3.9/site-packages/geopandas-0.8.1-py3.9.egg
> /Users/me/Library/Application Support/QGIS/QGIS3/profiles/default/python
> /Users/me/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/ee_plugin/extlibs
> /Users/me/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/kmltools/libs
> /Users/me/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/SentinelHub/external/requests_oauthlib-1.3.1-py2.py3-none-any.whl

**Comment by Fanchengyan** [comment]
Hi @oksoil 

You need to install Python dependencies before using GeoSAM. Here's the related document:  https://geo-sam.readthedocs.io/en/latest/installation.html#for-mac-linux-users
